consensus/src/payload_manager/quorum_store_payload_manager.rs

```diff
@@ -4,107 +4,62 @@
 
-    /// Check if the transactions corresponding are available. This is specific to payload
-    /// manager implementations. For optimistic quorum store, we only check if optimistic
-    /// batches are available locally.
-    fn check_payload_availability(&self, block: &Block) -> Result<(), BitVec>;
-
-    /// Get the transactions in a block's payload. This function returns a vector of transactions.
-    async fn get_transactions(
-        &self,
-        block: &Block,
-        block_voters: Option<BitVec>,
-    ) -> ExecutorResult<(Vec<SignedTransaction>, Option<u64>)>;
+pub trait TQuorumStoreCommitNotifier: Send + Sync {
+    fn notify(&self, block_timestamp: u64, batches: Vec<BatchInfo>);
 }
 
-/// A payload manager that directly returns the transactions in a block's payload.
-pub struct DirectMempoolPayloadManager {}
-
-impl DirectMempoolPayloadManager {
-    pub fn new() -> Self {
-        Self {}
-    }
+pub struct QuorumStoreCommitNotifier {
+    coordinator_tx: Sender<CoordinatorCommand>,
 }
 
-#[async_trait]
-impl TPayloadManager for DirectMempoolPayloadManager {
-    fn notify_commit(&self, _block_timestamp: u64, _payloads: Vec<Payload>) {}
-
-    fn prefetch_payload_data(&self, _payload: &Payload, _timestamp: u64) {}
-
-    fn check_payload_availability(&self, _block: &Block) -> Result<(), BitVec> {
-        Ok(())
+impl QuorumStoreCommitNotifier {
+    pub fn new(coordinator_tx: Sender<CoordinatorCommand>) -> Self {
+        Self { coordinator_tx }
     }
+}
 
-    async fn get_transactions(
-        &self,
-        block: &Block,
-        _block_signers: Option<BitVec>,
-    ) -> ExecutorResult<(Vec<SignedTransaction>, Option<u64>)> {
-        let Some(payload) = block.payload() else {
-            return Ok((Vec::new(), None));
-        };
+impl TQuorumStoreCommitNotifier for QuorumStoreCommitNotifier {
+    fn notify(&self, block_timestamp: u64, batches: Vec<BatchInfo>) {
+        let mut tx = self.coordinator_tx.clone();
 
-        match payload {
-            Payload::DirectMempool(txns) => Ok((txns.clone(), None)),
-            _ => unreachable!(
-                "DirectMempoolPayloadManager: Unacceptable payload type {}. Epoch: {}, Round: {}, Block: {}",
-                payload,
-                block.block_data().epoch(),
-                block.block_data().round(),
-                block.id()
-            ),
+        if let Err(e) = tx.try_send(CoordinatorCommand::CommitNotification(
+            block_timestamp,
+            batches,
+        )) {
+            warn!(
+                "CommitNotification failed. Is the epoch shutting down? error: {}",
+                e
+            );
         }
     }
 }
 
 /// A payload manager that resolves the transactions in a block's payload from the quorum store.
 pub struct QuorumStorePayloadManager {
     batch_reader: Arc<dyn BatchReader>,
-    coordinator_tx: Sender<CoordinatorCommand>,
+    commit_notifier: Box<dyn TQuorumStoreCommitNotifier>,
```

Comment:
why we need a trait now?

Reply:
I wanted to write some tests. Will do eventually.